modules/Bio/EnsEMBL/Compara/Production/DnaFragChunk.pm

```diff
@@ -133,56 +130,11 @@ sub fetch_masked_sequence {
   return undef unless($self->dnafrag);
   return undef unless($self->dnafrag->genome_db);
   return undef unless(my $dba = $self->dnafrag->genome_db->db_adaptor);
-  my $seq;
-  $dba->dbc->prevent_disconnect( sub {
-      $seq = $self->_fetch_masked_sequence();
-  } );
-  return $seq;
-}
-
-sub _fetch_masked_sequence {
-  my $self = shift;
-
-  return undef unless(my $slice = $self->slice());
-
-  my $seq;
-  my $id = $self->display_id;
-  my $masking_options;
-  my $starttime = time();
 
-  if(defined($self->masking_options)) {
-    $masking_options = eval($self->masking_options);
-    my $logic_names = $masking_options->{'logic_names'};
-    my $soft_masking = $masking_options->{'default_soft_masking'} // 1;
-    #printf("getting %s masked sequence...\n", $soft_masking ? 'SOFT' : 'HARD');
-
-    my $masked_slice = $slice->get_repeatmasked_seq($logic_names, $soft_masking, $masking_options);
-    $seq = Bio::PrimarySeq->new( -id => $id, -seq => $masked_slice->seq);
-  }
-  else {  # no masking options set, so get unmasked sequence
-    #print "getting UNMASKED sequence...\n";
-    $seq = Bio::PrimarySeq->new( -id => $id, -seq => $slice->seq);
-  }
-
-  #print ((time()-$starttime), " secs\n");
-
-  #print STDERR "sequence length : ",$seq->length,"\n";
-  $seq = $seq->seq;
-
-  if (defined $masking_options) {
-    foreach my $ae (@{$slice->get_all_AssemblyExceptionFeatures}) {
-      next unless (defined $masking_options->{"assembly_exception_type_" . $ae->type});
-      my $length = $ae->end - $ae->start + 1;
-      if ($masking_options->{"assembly_exception_type_" . $ae->type} == 0) {
-        my $padstr = 'N' x $length;
-        substr ($seq, $ae->start, $length) = $padstr;
-      } elsif ($masking_options->{"assembly_exception_type_" . $ae->type} == 1) {
-        my $padstr = lc substr ($seq, $ae->start, $length);
-        substr ($seq, $ae->start, $length) = $padstr;
-      }
-    }
-  }
+  #printf("getting %s masked sequence...\n", $self->masking ? 'SOFT' : 'HARD');
```

Comment:
`$self->masking` is a string, right?
```suggestion
  #printf("getting %s masked sequence...\n", $self->masking);
```

Reply:
In fact, if I recall correctly, it can be _undef_, `'soft'` or `'hard'`, so maybe it would be better to do:
```suggestion
  printf("getting %smasked sequence...\n", $self->masking ? $self->masking . ' ' : 'un') if $self->{debug};
```